src/Bicep.Core/Intermediate/Expression.cs

```diff
@@ -494,7 +494,8 @@ public record DeclaredResourceExpression(
     ImmutableArray<ResourceDependencyExpression> DependsOn,
     Expression? Description = null,
     ArrayExpression? RetryOn = null,
-    ArrayExpression? WaitUntil = null
+    ArrayExpression? WaitUntil = null,
+    ArrayExpression? OnlyIfNotExists = null
```

Comment:
Rather than adding a new expression for each decorator, could we make this generic by using a dictionary (with the "key" being the name of the decorator).

```suggestion
    ImmutableDictionary<string, ArrayExpression>? DecoratorConfig
```